jdaviz/configs/imviz/tests/test_orientation.py

```diff
@@ -1,31 +1,149 @@
 
-        # wcs_use_affine should revert/default to True
+        # wcs_use_affine should revert/default to True when change back to Pixels.
+        lc_plugin.link_type = 'Pixels'
         assert lc_plugin.wcs_use_affine is True
+        assert self.imviz.get_link_type("has_wcs_1[SCI,1]", "has_wcs_2[SCI,1]") == "pixels"
+
+        assert self.imviz.get_link_type("has_wcs_1[SCI,1]", "has_wcs_1[SCI,1]") == "self"
+
+        with pytest.raises(ValueError, match=".*combo not found"):
+            self.imviz.get_link_type("has_wcs_1[SCI,1]", "foo")
 
-        # adding markers should disable changing linking from both UI and API
-        assert lc_plugin.need_clear_astrowidget_markers is False
+    def test_astrowidgets_markers_disable_relinking(self):
+        lc_plugin = self.imviz.plugins['Orientation']
+        lc_plugin.link_type = 'Pixels'
+
+        # Adding markers should disable changing linking from both UI and API.
+        assert lc_plugin._obj.need_clear_astrowidget_markers is False
         tbl = Table({'x': (0, 0), 'y': (0, 1)})
         self.viewer.add_markers(tbl, marker_name='xy_markers')
 
-        assert lc_plugin.need_clear_astrowidget_markers is True
+        assert lc_plugin._obj.need_clear_astrowidget_markers is True
         with pytest.raises(ValueError, match="cannot change linking"):
-            lc_plugin.link_type.selected = 'WCS'
-        assert lc_plugin.link_type.selected == 'Pixels'
+            lc_plugin.link_type = 'WCS'
+        assert lc_plugin.link_type == 'Pixels'
+
+        lc_plugin._obj.vue_reset_astrowidget_markers()
+
+        assert lc_plugin._obj.need_clear_astrowidget_markers is False
+        lc_plugin.link_type = 'WCS'
+
+    def test_markers_plugin_recompute_positions_pixels_to_wcs(self):
+        lc_plugin = self.imviz.plugins['Orientation']
+        lc_plugin.link_type = 'Pixels'
+
+        # Blink to second image, if we have to.
+        if self.viewer.top_visible_data_label != "has_wcs_2[SCI,1]":
+            self.viewer.blink_once()
+
+        label_mouseover = self.imviz.app.session.application._tools['g-coords-info']
+        mp = self.imviz.plugins['Markers']
+        mp._obj.plugin_opened = True
+
+        # (0, 0) on second image.
+        label_mouseover._viewer_mouse_event(
+            self.viewer, {'event': 'mousemove', 'domain': {'x': 1, 'y': 0}})
+        mp._obj._on_viewer_key_event(self.viewer, {'event': 'keydown', 'key': 'm'})
+
+        # (0, 0) on first image.
+        self.viewer.blink_once()
+        label_mouseover._viewer_mouse_event(
+            self.viewer, {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
+        mp._obj._on_viewer_key_event(self.viewer, {'event': 'keydown', 'key': 'm'})
+
+        lc_plugin.link_type = 'WCS'
+
+        # Both marks stay the same in sky, so this means X and Y w.r.t. reference
+        # same on both entries.
+        # 0.25 offset probably introduced by fake WCS layer.

```

Comment:
I still need to parse what is expected here.... but do you think this 0.25 is expected or a bug?

Reply:
Hard to tell with the fake WCS layer in play. Maybe Brett knows?

Reply:
I think this is the remaining thing that needs an answer before we merge?